molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/controller/MappingServiceController.java

```diff
@@ -244,6 +251,88 @@ public String removeEntityMapping(@RequestParam String mappingProjectId, String
 		return "redirect:/menu/main/mappingservice/mappingproject/" + mappingProjectId;
 	}
 
+	@RequestMapping(value = "/validateAttrMapping", method = RequestMethod.POST)
+	@ResponseBody
+	public AttributeMappingValidationReport validateAttributeMapping(
+			@Valid @RequestBody MappingServiceRequest mappingServiceRequest)
+	{
+		String targetEntityName = mappingServiceRequest.getTargetEntityName();
+		EntityMetaData targetEntityMeta = dataService.getEntityMetaData(targetEntityName);
+
+		String targetAttributeName = mappingServiceRequest.getTargetAttributeName();
+		AttributeMetaData targetAttr = targetEntityMeta.getAttribute(targetAttributeName);
+		if (targetAttr == null)
+		{
+			throw new UnknownAttributeException("Unknown attribute [" + targetAttributeName + "]");
+		}
+
+		String algorithm = mappingServiceRequest.getAlgorithm();
+		Long offset = mappingServiceRequest.getOffset();
+		Long num = mappingServiceRequest.getNum();
+		Query query = new QueryImpl().offset(offset.intValue()).pageSize(num.intValue());
+		String sourceEntityName = mappingServiceRequest.getSourceEntityName();
+		Iterable<Entity> sourceEntities = dataService.findAll(sourceEntityName, query);
+
+		long total = dataService.count(sourceEntityName, new QueryImpl());
+		long nrSuccess = 0, nrErrors = 0;
+		Map<String, String> errorMessages = new LinkedHashMap<String, String>();
+		for (AlgorithmEvaluation evaluation : algorithmService.applyAlgorithm(targetAttr, algorithm, sourceEntities))
```

Comment:
Line 264 checks if the targetAttribute is null i.e. not there. The null pointer that is thrown is caused by an attribute that does not exist in the source entity